pallets/oracle/src/lib.rs

```diff
@@ -0,0 +1,511 @@
+			log::trace!("Request: {}", req);
+
+			let wait = Timestamp::from_unix_millis(FETCH_TIMEOUT_PERIOD);
+			let response_body = http_json_post(&geth_uri, req.as_bytes(), Some(wait));
+			let response_body = if let Ok(response) = response_body {
+				response
+			} else {
+				return Err("Failed to get response from Ethereum.")
+			};
+
+			let response = String::from_utf8(response_body).map_err(|_| "Invalid response")?;
+			log::trace!("Response: {}", response);
+
+			let v: Value =
+				serde_json::from_str(&response).map_err(|_| "Invalid response - json parse")?;
+			let result = v["result"].as_str().ok_or("Invalid response - no result")?; // Get the latest block number of the Ethereum Blockchain
+			let data = hex::decode(&result[2..]).map_err(|_| "Invalid response - invalid data")?;
+			log::trace!("data: {:?}", data);
+			let data = ethabi::decode(
+				//https://docs.chain.link/docs/data-feeds/price-feeds/api-reference/#latestrounddata
+				&[ParamType::Tuple(vec![
+					ParamType::Uint(80),  // uint80 roundId
+					ParamType::Int(256),  // int256 answer
+					ParamType::Uint(256), // uint256 startedAt
+					ParamType::Uint(256), // uint256 updatedAt
+					ParamType::Uint(80),  // uint80 answeredInRound
+				])],
+				&data,
+			)
+			.map_err(|_| "Invalid response")?;
+
+			let tuple = data[0].clone().into_tuple().ok_or_else(|| "Invalid tuple")?;
+			let round_id = tuple[0].clone().into_uint().ok_or_else(|| "Invalid roundId")?;
+			let price = tuple[1].clone().into_int().ok_or_else(|| "Invalid token")?;
+			let started_at = tuple[2].clone().into_uint().ok_or_else(|| "Invalid startedAt")?;
+			let updated_at = tuple[3].clone().into_uint().ok_or_else(|| "Invalid updatedAt")?;
+			let answered_in_round =
+				tuple[4].clone().into_uint().ok_or_else(|| "Invalid answeredInRound")?;
+
+			/*
+			The following data validations have been inspired by:
+			- https://github.com/code-423n4/2021-08-notional-findings/issues/92
+			- https://github.com/code-423n4/2022-02-hubble-findings/issues/123
+			- https://ethereum.stackexchange.com/questions/133890/chainlink-latestrounddata-security-fresh-data-check-usage
+			and other similar reports: https://github.com/search?q=latestrounddata+validation&type=issues
+			*/
+			ensure!(round_id.gt(&U256::from(0)), "Price from oracle is 0");
+			ensure!(price > U256::from(0), "Price from oracle is <= 0");
+			ensure!(!updated_at.is_zero(), "UpdateAt = 0. Incomplete round.");
+			ensure!(!answered_in_round.is_zero(), "AnsweredInRound from oracle is 0");
+			ensure!(answered_in_round.ge(&round_id), "Stale price");
+
+			log::trace!("New price: {}", price);
+
+			// -- Sign using any account
+			Signer::<T, T::AuthorityId>::any_account()
+				.send_unsigned_transaction(
+					|account| OraclePrice {
+						round_id,
+						price,
+						started_at,
+						updated_at,
+						answered_in_round,
+						block_number,
+						public: account.public.clone(),
```

Comment:
pretty sure clone is not needed here

Reply:
I tried to not use it, but I get
```rust
move occurs because `account.public` has type `<T as frame_system::offchain::SigningTypes>::Public`, which does not implement the `Copy` trait
```

Reply:
👍